bin/sirius-script-si-set-loco-strengths.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python-sirius
+"""Script to set LOCO-fitted strengths to SI and correct coupling."""
+
+import os
+import sys
+
+import numpy as np
+from mathphys.functions import load_pickle
+from apsuite.commisslib.apply_strengths import (
+    SetOpticsMode,
+    SISetTrimStrengths,
+)
+
+# Achromatic skew quadrupole variations to adjust betatron coupling,
+# it is -(v1 + v4)/np.sqrt(2) where v1 and v4 are the first and fourth
+# left-singular vectors of jacobian matrix calculated with
+# apsuite.optics.coupling_correction.calc_jacobian_matrix() using the
+# nominal SIRIUS lattice
+ACHROM_QS_ADJ = 1e-4 * np.array(
+    [
+        -0.53435125,
+        1.83818311,
+        1.98691151,
+        1.75361165,
+        1.56175837,
+        -0.88710175,
+        -1.15014152,
+        -2.18259391,
+        -0.53435125,
+        1.83818311,
+        1.98691151,
+        1.75361165,
+        1.56175837,
+        -0.88710175,
+        -1.15014152,
+        -2.18259391,
+        -0.53435125,
+        1.83818311,
+        1.98691151,
+        1.75361165,
+        1.56175837,
+        -0.88710175,
+        -1.15014152,
+        -2.18259391,
+        -0.53435125,
+        1.83818311,
+        1.98691151,
+        1.75361165,
+        1.56175837,
+        -0.88710175,
+        -1.15014152,
+        -2.18259391,
+        -0.53435125,
+        1.83818311,
+        1.98691151,
+        1.75361165,
+        1.56175837,
+        -0.88710175,
+        -1.15014152,
+        -2.18259391,
+    ]
+)
+
+
+def ask_yes_no(prompt):
+    """Prompt user for yes/no input. Default to yes."""
+    print(f"{prompt} [Y/N]: ", end="", flush=True)
+    try:
+        response = input().strip().lower()
+    except KeyboardInterrupt:
+        print("\nAborted.")
+        sys.exit(1)
+    return response in ("", "y", "yes")
+
+
+def prompt_apply_then_check(apply_func):
+    """Prompt to accept changes and apply, then prompt to reapply if needed."""
+    if not ask_yes_no("Accept changes?"):
+        sys.exit(0)
+    apply_func()
+    if ask_yes_no("Check if applied?"):
+        print("Reapplying...")
```

Comment:
why `"Reapplying"`? isn't this the first time the strengthss will be applied?

Reply:
The changes are first applied by `apply_func`, in line 80. These functions have this `print_change` arg which, when true, prints out the strengths that will be applied to the magnets in order to reach the desired (fitted) strengths. If zeros are printed, we know it's all been set.  In other words, as in the original notebook, the `apply_func` is also used to check if changes have been set correctly.

Reply:
oh, right, only now I saw the apply_func just above. But is there a wait time between setting the values and printing the comparison on the apply_func function? because if there isn't, the user will always think it didn't apply and will call apply_func again, which is not optimal.